extensions/ql-vscode/src/extension.ts

```diff
@@ -549,13 +555,57 @@ async function activateWithInstalledDistribution(
         progress: ProgressCallback,
         token: CancellationToken,
         uri: Uri | undefined
-      ) => await compileAndRunQuery(false, uri, progress, token),
+      ) => await compileAndRunQuery(false, uri, progress, token, undefined),
       {
         title: 'Running query',
         cancellable: true
       }
     )
   );
+  interface DatabaseQuickPickItem extends QuickPickItem {
+    databaseItem: DatabaseItem;
+  }
+  ctx.subscriptions.push(
+    commandRunnerWithProgress(
+      'codeQL.runQueryOnMultipleDatabases',
+      async (
+        progress: ProgressCallback,
+        token: CancellationToken,
+        uri: Uri | undefined
+      ) => {
+        const quickPickItems = dbm.databaseItems.map<DatabaseQuickPickItem>(dbItem => (
+          {
+            databaseItem: dbItem,
+            label: dbItem.name,
+            description: dbItem.language,
+          }
+        ));
+        /**
+         * Databases that were selected in the quick pick menu.
+         */
+        const quickpick = await window.showQuickPick<DatabaseQuickPickItem>(
+          quickPickItems,
+          { canPickMany: true }
+        );
+        if (quickpick !== undefined) {
+          for (const item of quickpick) {
+            try {
+              await compileAndRunQuery(false, uri, progress, token, item.databaseItem);
+            } catch (error) {
+              // Skip databases that are incompatible with the query, e.g. using a different language.
+              void helpers.showAndLogErrorMessage(`Skipped database '${item.label}'. ${error}`);
```

Comment:
If multiple databases give an error, this pops up one dialog per database, which can get a bit overwhelming.

I suggest creating a variable outside the loop that holds a list/array of errors (strings). Every time you catch an error, add the error to the list instead of showing the error right away.
Then outside the loop, pop a single dialog that logs all the errors as a single error, with some suitable header message. You can use `.join('\n')` for example to combine the array of strings with newlines in between.

Reply:
As discussed offline, turns out the popup boxes don't support Markdown 😢  So consider logging all the errors, but only creating one popup with the list of database names that failed.

Reply:
Thank you! I had a go at this in the latest commit. It turns out that `showAndLogErrorMessage` completely ignores newlines (even in the logs), so I've used `showAndLogWarningMessage` instead to make things a bit more readable...

I ended up creating two separate variables (`skippedDatabases` and `errors`) for tracking the errors, which feels a bit messy? Let me know if there's a neater/more compact way of doing this!